src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/clients/OrderlyWebIndirectClient.kt

```diff
@@ -33,11 +41,17 @@ class OrderlyWebIndirectClientRedirectActionBuilder: RedirectActionBuilder
 {
     override fun redirect(context: WebContext) : RedirectAction
     {
-        //Attach the originally requested url to the redirect url as a query string parameter, so we can redirect
-        //there once authenticated via the 'landing page'
-        val requestedUrl = context.fullRequestURL
-        val encodedUrl = URLEncoder.encode(requestedUrl, "utf-8")
-        return RedirectAction.redirect(AppConfig()["app.url"] + "/weblogin?requestedUrl=" + encodedUrl)
+        val profile = CommonProfile().apply { id = "anon" }
+
+        val permissions = OrderlyAuthorizationRepository()
+                .getPermissionsForUser("anon")
+
+        profile.orderlyWebPermissions = permissions.plus(ReifiedPermission("reports.read",
+                Scope.Specific("report", "minimal")))
+
+        val manager = ProfileManager<CommonProfile>(context)
+        manager.save(true, profile, false)
+        return RedirectAction.redirect(context.fullRequestURL)
```

Comment:
So if there's no user profile in the session, instead of redirecting the user to the login page, just add the anon profile to the session. The user can still login in manually by directly visiting `appUrl/weblogin`.

Reply:
Would have to do this conditional on fine grained auth being true. If fine grained auth is off, this approach would totally obviate the need for a login - the anon user could access everything. There might be a use case for that in internal instances not on the public internet, but I guess would introduce another config flag "allow_anon" or something.